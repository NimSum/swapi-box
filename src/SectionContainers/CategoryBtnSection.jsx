import React, { Component } from 'react'
import { PlanetsBtn, PeopleBtn, VehiclesBtn, ViewFavoritesBtn } from '../components/Buttons/Buttons';

export class CategoryBtnSection extends Component {
  constructor() {
    super();
    this.state = {
      activeBtn: '',
    }
  }

  setActiveBtn = e => {
    this.setState({ activeBtn: e.target.name });
    this.props.changeCategory(e.target.name);
  }

  render() {
    const { activeBtn } = this.state;

    return (
      <section className="categories-btn-container">
        <div className="btn-container">
          < PlanetsBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } />
          < PeopleBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } />
          < VehiclesBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } />
          < ViewFavoritesBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } 
            favoriteCount={ this.props.favoriteCount } />
        </div>
      </section>
    )
  }
}

export default CategoryBtnSection
