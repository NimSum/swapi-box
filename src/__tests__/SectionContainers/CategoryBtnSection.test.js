import React from 'react';
import { shallow } from 'enzyme';
import CategoryBtnSection from '../../SectionContainers/CategoryBtnSection';

describe('Category Button Section', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      < CategoryBtnSection />
    );
  });

  const mockEvent = {
    target: { name: "people"}
  }

  it('Should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should set state active button', () => {
    expect(wrapper.state().activeBtn).toEqual('');
    wrapper.instance().setActiveBtn(mockEvent);
    expect(wrapper.state().activeBtn).toEqual('people');
  })

})