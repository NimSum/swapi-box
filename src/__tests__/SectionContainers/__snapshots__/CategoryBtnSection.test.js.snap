// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Button Section Should match component snapshot 1`] = `
<section
  className="categories-btn-container"
>
  <div
    className="btn-container"
  >
    <PlanetsBtn
      activeBtn=""
      setActive={[Function]}
    />
    <PeopleBtn
      activeBtn=""
      setActive={[Function]}
    />
    <VehiclesBtn
      activeBtn=""
      setActive={[Function]}
    />
    <ViewFavoritesBtn
      activeBtn=""
      setActive={[Function]}
    />
  </div>
</section>
`;
